Sources/FluentPostgreSQL/PostgreSQLDriver.swift

```diff
@@ -56,23 +58,7 @@ public class PostgreSQLDriver: Fluent.Driver {
 
         switch query.action {
         case .create:
-            // fetch the last inserted value
-            do {
-                let lastval = try _execute("SELECT LASTVAL();", [], connection)
-                // check if it contains an id
-                if let id = lastval[0, "lastval"]?.int {
-                    // return the id to fluent to
-                    // be the model's new id
-                    return .number(.int(id))
-                }
-            }
-            catch PostgreSQL.DatabaseError.invalidSQL(message: _) {
-                // When receiving a invalidSQL error, there is basically no LASTVAL, so ignore.
-            }
-
-            // no id found, return whatever
-            // the results are
-            return result
+            return (result[idKey]?.array as? [Node])?.first ?? result
```

Comment:
I guess one way to future proof this bit would be to return the first item if the array's size is 1, and return the entire array if it is higher than that (if genuinely a vector rather than a scalar was inserted). That, or place a precondition to state that the max size expected is 1, i.e. that a single record was inserted as there is to my understanding no way to insert a batch of records at the moment through the Fluent API?